core/database/arango_db_manager.py

```
@@ -0,0 +1,252 @@
+        Get a connection from the pool.
+        
+        Args:
+            timeout: Maximum time to wait for a connection
+            
+        Returns:
+            Database connection from the pool
+        """
+        try:
+            return self.available_connections.get(timeout=timeout)
+        except Empty:
+            raise TimeoutError(f"Could not get database connection within {timeout} seconds")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)